hail/src/main/scala/is/hail/expr/ir/functions/ArrayFunctions.scala

```diff
@@ -316,8 +316,8 @@ object ArrayFunctions extends RegistryFunctions {
         val x = r.mb.newLocal[Double]
         val y = r.mb.newLocal[Double]
 
-        val a1 = v1.asInstanceOf[Code[Long]]
-        val a2 = v2.asInstanceOf[Code[Long]]
+        val a1 = v1.tcode[Long]
```

Comment:
Where does the implicit TypeInfo[Long] come from?

Reply:
LongInfo is an implicit object in the asm4s package.  There are implicits that should handle any concrete type.  You only need an implicit TypeInfo parameter for a type variable like T.

Reply:
Thanks!